cmd/kubeadm/app/phases/certs/certs.go

```diff
@@ -286,9 +286,10 @@ type certKeyLocation struct {
 
 // SharedCertificateExists verifies if the shared certificates - the certificates that must be
 // equal across control-plane nodes: ca.key, ca.crt, sa.key, sa.pub + etcd/ca.key, etcd/ca.crt if local/stacked etcd
+// A missing ca.key (external CA) is a special case that causes a warning instead of an error.
 func SharedCertificateExists(cfg *kubeadmapi.ClusterConfiguration) (bool, error) {
 
-	if err := validateCACertAndKey(certKeyLocation{cfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName, "", "CA"}); err != nil {
+	if err := validateCACertAndKeyWarn(certKeyLocation{cfg.CertificatesDir, kubeadmconstants.CACertAndKeyBaseName, "", "CA"}); err != nil {
```

Comment:
i guess we should have warnings instead of errors for the front-proxy's CA key and etcd's CA key too.
if the CA is external users may choose to generate everything externally and place the above CA certificates only without the keys. unlike the root singing CA key, these keys are not needed during runtime.

Reply:
@fabriziopandini i have update the PR.
instead of adding a new function validateCACertAndKey is re-purposed to throw a warning.